File: python/valid_parantheses.py
def isValid(s):
    """
    :type s: str
    :rtype: bool
    """
    stack = []
    para_map = {"}": "{", "]": "[", ")": "("}
    for char in s:
        if char in para_map:
            top_ele = stack.pop() if stack else "#"
            if para_map[char] != top_ele:
                return False
        else:
            stack.append(char)
    return not stack

# Test
checkValidPara = isValid("([])")
assert checkValidPara == True

checkValidPara = isValid("([][)")
assert checkValidPara == False

File: javascript/valid_parantheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValidPara = function (s) {
  let stack = [];
  let para_map = { "}": "{", "]": "[", ")": "(" };
  for (let i = 0; i < s.length; i++) {
    if (!para_map[s[i]]) {
      stack.push(s[i]);
    } else {
      let top_element = stack.pop();
      if (top_element !== para_map[s[i]]) return false;
    }
  }
  return stack.length == 0;
};

module.exports = { isValidPara };
